src/tools/perf/lib/libperf.c

```diff
@@ -1717,66 +1863,93 @@ static void* ucx_perf_thread_run_test(void* arg)
             goto out;
         }
     }
-#pragma omp master
-    {
-        /* Assuming all threads are fairly treated, reporting only tid==0
-            TODO: aggregate reports */
-        ucx_perf_calc_result(perf, result);
-        rte_call(perf, report, result, perf->params.report_arg, 1);
-    }
+
+    ucx_perf_calc_result(perf, result);
 
 out:
     return &statuses[tid];
 }
 
+static void ucx_perf_thread_report_aggregated_results(ucx_perf_context_t *perf)
+{
+    ucx_perf_thread_context_t* tctx = perf->ucp.tctx;  /* all the thread contexes on perf */
+    unsigned i, thread_count        = perf->params.thread_count;
+    double lat_sum_momemt_avegare   = 0.0;
+    double lat_sum_total_avegare    = 0.0;
+    double lat_sum_typical_avegare  = 0.0;
+    ucx_perf_result_t agg_result;
+
+    agg_result.iters        = tctx[0].result.iters;
+    agg_result.bytes        = tctx[0].result.bytes;
+    agg_result.elapsed_time = tctx[0].result.elapsed_time;
+
+    agg_result.bandwidth.total_average  = 0.0;
+    agg_result.bandwidth.typical        = 0.0; // Undefined
+    agg_result.latency.moment_average   = 0.0;
+    agg_result.latency.total_average    = 0.0;
+    agg_result.msgrate.moment_average   = 0.0;
+    agg_result.msgrate.total_average    = 0.0;
+    agg_result.msgrate.typical          = 0.0; // Undefined
+
+    for (i = 0; i < thread_count; i++) {
+        agg_result.bandwidth.total_average  += tctx[i].result.bandwidth.total_average;
+
+        agg_result.msgrate.moment_average   += tctx[i].result.msgrate.moment_average;
+        agg_result.msgrate.total_average    += tctx[i].result.msgrate.total_average;
+
+        lat_sum_momemt_avegare              += tctx[i].result.latency.moment_average;
+        lat_sum_total_avegare               += tctx[i].result.latency.total_average;
+        lat_sum_typical_avegare             += tctx[i].result.latency.typical;
+    }
+
+    agg_result.latency.moment_average = lat_sum_momemt_avegare  / thread_count;
+    agg_result.latency.total_average  = lat_sum_total_avegare   / thread_count;
+    agg_result.latency.typical        = lat_sum_typical_avegare / thread_count;
+
+    /* when running with mutliple threads, the moment average value is
+     * undefined since we don't capture the values of the last iteration */
+    agg_result.bandwidth.moment_average = agg_result.bandwidth.total_average;
```

Comment:
it should be 0.0

Reply:
IMHO if a user sees 0.0 at the output of a BW/latency/msg_rate column of a ucx performance test, he/she would definitely have questions about this...

Reply:
so make it "N/A" or even NaN (=not a number), IMHO it's better than displaying misleading perf result

Reply:
you mean to print a string there? 
would have to really refactor print_progress
https://github.com/openucx/ucx/blob/master/src/tools/perf/perftest.c#L231
all are numbers there and can't tell if more than one thread was used since only the main thread calls this function at the final stage.

Reply:
yes, need to change the print function to accept strings for typica and average parts and prepare those strings before - either as real numbers or n/a